Core/clim-basic/geometry/region-predicates.lisp

```diff
@@ -83,21 +134,80 @@
                   (when (and (> y2 y)
                              (> (is-left x1 y1 x2 y2 x y) 0))
                     (incf wn))
                   (when (and (<= y2 y)
                              (< (is-left x1 y1 x2 y2 x y) 0))
                     (decf wn))))
-            self)
+            region)
            (not (zerop wn))))))
 
+(defmethod region-contains-position-p ((self rectangle) x y)
+  (multiple-value-bind (x1 y1 x2 y2)
+      (rectangle-edges* self)
+    (and (coordinate-between* x1 x x2)
+         (coordinate-between* y1 y y2))))
+
+(defmethod region-contains-position-p ((self ellipse) x y)
+  (nest
+   (multiple-value-bind (cx cy) (ellipse-center-point* self))
+   (let ((alpha (or (ellipse-start-angle self) 0))
+         (omega (or (ellipse-end-angle self) (* 2 pi)))
+         (polar->screen (polar->screen self)))
+     (and (angle-contains-point-p alpha omega (- x cx) (- y cy))
+          (nest (multiple-value-bind (polar-x polar-y)
+                    (untransform-position polar->screen x y))
+                ;; FIXME we don't need to factor the additonal epsilon
+                ;; but rotated elliptoids are naively rendered in clx.
+                (multiple-value-bind (polar-dx polar-dy)
+                    (untransform-distance polar->screen 1 1))
+                (coordinate<= (+ (square polar-x) (square polar-y))
+                              (+ 1 (square polar-dx) (square polar-dy))))))))
+
+
+;;   REGION-INTERSECTS-REGION-P region1 region2
+;;
+;;        Returns nil if region-intersection of the two regions region1 and
+;;        region2 would be +nowhere+; otherwise, it returns t.
+;;
+;;        aka region1 und region2 sind nicht disjunkt  aka AB /= 0
+;;
+
+;; "generic" version
+(defmethod region-intersects-region-p ((a region) (b region))
+  (not (region-equal +nowhere+ (region-intersection a b))))
+
+(defmethod region-intersects-region-p :around ((a bounding-rectangle) (b bounding-rectangle))
+  (multiple-value-bind (x1 y1 x2 y2) (bounding-rectangle* a)
+    (multiple-value-bind (u1 v1 u2 v2) (bounding-rectangle* b)
+      (cond ((and (<= u1 x2) (<= x1 u2)
+                  (<= v1 y2) (<= y1 v2))
+             (call-next-method))
+            (t
+             nil)))))
+
+(defmethod region-intersects-region-p ((a standard-rectangle) (b standard-rectangle))
+  (declare (ignorable a b))
+  ;; for rectangles, the bounding rectangle test is correct, so if we
+  ;; wind up here, we just can return T.
+  t
+  ;;(multiple-value-bind (x1 y1 x2 y2) (rectangle-edges* a)
+  ;;  (multiple-value-bind (u1 v1 u2 v2) (rectangle-edges* b)
+  ;;    (and (<= u1 x2) (<= x1 u2)
+  ;;         (<= v1 y2) (<= y1 v2))))
+  )
+
 
 ;;   REGION-CONTAINS-REGION-P region1 region2
 ;;
-;;        Returns t if all points in the region region2 are members of the
-;;        region region1; otherwise, it returns nil.
+;;        Returns T if all points in the region are members of the
+;;        region B; otherwise, it returns nil.
```

Comment:
I think the previous version of the comment made more sense since it referred to the parameters.

Reply:
fixed